source/fab/tools/tool.py

```diff
@@ -91,11 +91,26 @@ def exec_name(self) -> str:
         ''':returns: the name of the executable.'''
         return self._exec_name
 
+    def change_exec_name(self, exec_name: str):
```

Comment:
I'm concerned that this is open to abuse or mistaken use. Would it be better to provide a method which returns the command line without the executable which a wrapping class could then use to prepend with its own executable and arguments?

Reply:
After some thinking, I have added a #TODO for #370. Main reason is that I nearly did exactly that when allowing linker to be 'wrapped'. Since I might have modified related functions, I'll address this issue as part of the branch `linker_wrapper_new` to avoid fighting with conflicts (it's around 6 PRs ahead of this).